micrometer-observation-test/src/main/java/io/micrometer/observation/tck/ObservationRegistryCompatibilityKit.java

```diff
@@ -298,21 +298,47 @@ void runnableShouldBeScoped() {
         Runnable runnable = () -> assertThat(registry.getCurrentObservation()).isSameAs(observation);
         observation.scoped(runnable);
         assertThat(registry.getCurrentObservation()).isNull();
+        assertThat(observation.getContext().getError()).isEmpty();
     }
 
     @Test
     void errorShouldBeReportedOnFailingScopedRunnable() {
         registry.observationConfig().observationHandler(c -> true);
-        Observation.Context context = new Observation.Context();
-        Observation observation = Observation.start("myObservation", context, registry);
+        Observation observation = Observation.start("myObservation", registry);
         RuntimeException error = new RuntimeException("simulated");
 
         Runnable runnable = () -> {
+            assertThat(registry.getCurrentObservation()).isSameAs(observation);
             throw error;
         };
         assertThatThrownBy(() -> observation.scoped(runnable)).isSameAs(error);
-        assertThat(context.getError()).containsSame(error);
         assertThat(registry.getCurrentObservation()).isNull();
+        assertThat(observation.getContext().getError()).containsSame(error);
+    }
+
+    @Test
+    void checkedRunnableShouldBeScoped() throws Throwable {
+        registry.observationConfig().observationHandler(c -> true);
+        Observation observation = Observation.start("myObservation", registry);
+        Observation.CheckedRunnable runnable = () -> assertThat(registry.getCurrentObservation()).isSameAs(observation);
+        observation.scopedChecked(runnable);
+        assertThat(registry.getCurrentObservation()).isNull();
+        assertThat(observation.getContext().getError()).isEmpty();
```

Comment:
I think it would be good to assert that the observation is not stopped by using `scopedChecked`.